hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1481,7 +1488,31 @@ func (r *HostedClusterReconciler) reconcileAutoscaler(ctx context.Context, hclus
 	return nil
 }
 
-func reconcileControlPlaneOperatorDeployment(deployment *appsv1.Deployment, hc *hyperv1.HostedCluster, image string, sa *corev1.ServiceAccount) error {
+func reconcileControlPlaneOperatorDeployment(deployment *appsv1.Deployment, hc *hyperv1.HostedCluster, image string, pullSecret []byte, sa *corev1.ServiceAccount) error {
+	releaseProvider := &releaseinfo.CachedProvider{
+		Inner: &releaseinfo.RegistryClientProvider{},
+		Cache: map[string]*releaseinfo.ReleaseImage{},
+	}
+	releaseInfo, err := releaseProvider.Lookup(context.TODO(), hc.Spec.Release.Image, pullSecret)
+	if err != nil {
+		return err
+	}
+	version, err := semver.Parse(releaseInfo.Version())
+	if err != nil {
+		return err
+	}
+	versionMajMin := fmt.Sprintf("%d.%d", version.Major, version.Minor)
+	pullSpec := "registry.ci.openshift.org/hypershift/hypershift"
+	var controlPlaneOperatorImage string
+	switch versionMajMin {
+	case "4.9":
```

Comment:
this case should likely be _>=4.9_

Reply:
I thought this could be a way to allowlist the version we support.

Reply:
hmm, it does work with 4.10 today (until 4.10 gets a new kube rebase). We could include "4.10" and remove it if it stops working, would that be better?

Reply:
sounds good